core/src/test/java/com/xmartlabs/base/core/observers/MaybeMainObserversWithErrorHandlingTest.java

```diff
@@ -0,0 +1,178 @@
+
+import org.junit.Test;
+
+import io.reactivex.Maybe;
+import io.reactivex.plugins.RxJavaPlugins;
+import timber.log.Timber;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+public class MaybeMainObserversWithErrorHandlingTest extends MainObserversWithErrorHandlingTest {
+  @Override
+  @SuppressWarnings("unchecked")
+  protected void setObserverSubscriber() {
+    RxJavaPlugins.setOnMaybeSubscribe((Maybe, MaybeObserver) ->
+        new MaybeObserverWithErrorHandling<>(MaybeObserver, getErrorHandlingAction()));
+  }
+
+  @Test
+  public void callsMaybeSubscribe() {
+    Maybe.just(new Object())
+        .doOnSubscribe(disposable -> Timber.tag(OBSERVABLE_SUBSCRIBE).i(OBSERVABLE_SUBSCRIBE))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_SUBSCRIBE).getTag(), equalTo(OBSERVABLE_SUBSCRIBE));
+  }
+
+  @Test
+  public void callsMaybeOnSuccessWhenOneObjectIsEmmited() {
+    Maybe.just(new Object())
+        .doOnSuccess(o -> Timber.tag(OBSERVABLE_SUCCESS).i(OBSERVABLE_SUCCESS))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_SUCCESS).getTag(), equalTo(OBSERVABLE_SUCCESS));
+  }
+
+  @Test
+  public void doesNotCallMaybeOnSuccessWhenNoObjectIsEmmited() {
+    Maybe.fromCallable(() -> null)
+        .doOnSuccess(o -> Timber.tag(OBSERVABLE_SUCCESS).i(OBSERVABLE_SUCCESS))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_SUCCESS).getTag(), equalTo(DEFAULT_NOT_FOUND_TREE_NODE));
+  }
+
+  @Test
+  public void doesNotCallMaybeOnCompleteWhenOneObjectIsEmmited() {
+    Maybe.just(new Object())
+        .doOnComplete(() -> Timber.tag(OBSERVABLE_ON_COMPLETE).i(OBSERVABLE_ON_COMPLETE))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_ON_COMPLETE).getTag(), equalTo(DEFAULT_NOT_FOUND_TREE_NODE));
+  }
+
+  @Test
+  public void callsMaybeOnCompleteWhenNoObjectOmited() {
+    Maybe.fromCallable(() -> null)
+        .doOnComplete(() -> Timber.tag(OBSERVABLE_ON_COMPLETE).i(OBSERVABLE_ON_COMPLETE))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_SUCCESS).getTag(), equalTo(OBSERVABLE_SUCCESS));
+  }
+
+  @Test
+  public void callsMaybeOnError() {
+    Maybe
+        .fromCallable(() -> {
+          throw new Exception(OBSERVABLE_ACTION_EXCEPTION);
+        })
+        .doOnError(throwable -> Timber.tag(OBSERVABLE_DO_ON_ERROR).i(OBSERVABLE_DO_ON_ERROR))
+        .subscribe(o -> {
+        });
+
+    assertThat(getLogTreeNodeWithTag(OBSERVABLE_DO_ON_ERROR).getTag(), equalTo(OBSERVABLE_DO_ON_ERROR));
+  }
+
+  @Test
+  public void doesNotCallOnSuccessWhenError() {
+    Maybe
+        .fromCallable(() -> {
+          throw new Exception(OBSERVABLE_ACTION_EXCEPTION);
+        })
+        .doOnSuccess(o -> Timber.e(OBSERVABLE_SUCCESS))
+        .subscribe(o -> {
```

Comment:
Won't be better to make this as an only line? I mean like `.subscribe(o -> {});`